libyul/optimiser/SimplificationRules.cpp

```diff
@@ -250,7 +250,8 @@ Expression Pattern::toExpression(shared_ptr<DebugData const> const& _debugData)
 		for (auto const& arg: m_arguments)
 			arguments.emplace_back(arg.toExpression(_debugData));
 
-		string name = util::toLower(instructionInfo(m_instruction).name);
+		//TODO:prevrandao: check if default EVM version is fine here.
+		string name = util::toLower(instructionInfo(m_instruction, EVMVersion()).name);
```

Comment:
This one looks like it is a problem.

Reply:
But since the usage in ``KnowledgeBase`` will vanish with https://github.com/ethereum/solidity/pull/13702, and at the other usage you have an evm version available, you should be able to pass it down here...
In the ``KnowledgeBase`` you can pass the default EVM version for now.

Reply:
I solved it by creating a method to return the EVM version when it is an EVM dialect or return the default EVM version otherwise. Please see here: https://github.com/ethereum/solidity/pull/13759/commits/66573a337157ba374c1d316909c7d45486f963c2